game/scripts/vscripts/TutorialGraph/Steps.ts

```diff
@@ -76,6 +105,49 @@ export const spawnAndKillUnit = (unitName: string, spawnLocation: Vector, visibl
     })
 }
 
+/**
+ * Creates a tutorial step that sets a unit's movement capability.
+ * @param getUnitFunc Function that returns a CDOTA_BaseNPC type.
+ * @param unitMoveCap UnitMoveCapability enum for setting move capability.
+ */
+export const setUnitMoveCapability = (getUnitFunc: () => CDOTA_BaseNPC, unitMoveCap: UnitMoveCapability) => {
+    return step((context, complete) => {
+        getUnitFunc().SetMoveCapability(unitMoveCap)
```

Comment:
couldnt we just use a doImmediate for this?

Reply:
Do you mean something like this from #48 :

`        tg.immediate(context => playerHero.HeroLevelUp(true)),
`

Reply:
Decided to implement this later since an immediate function exists in #48 that is awaiting approval of this PR